### PR TITLE
Added support to send string values as key for StatsD.

### PR DESCRIPTION
**Pull request by gmcoringa**
Hello,

Added support to send string values as part of the key for StatsD.

This is necessary as StatsD only support numbers, but you wish to monitor server status, for example Zookeeper.

This PR add two new settings for StatsD:
* ``stringValuesAsKey``: a boolean that enable this feature
* ``stringValueDefaultCount``: when enabled this setting is the count value for the attributes collected.

Fabiano
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/jmxtrans/jmxtrans/pull/363%23discussion_r45325132%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/363%23issuecomment-158023094%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/363%23issuecomment-158045975%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/363%23issuecomment-158023094%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20sent%20you%20some%20ideas%20on%20what%20I%20think%20is%20cleaner%20code%20via%20another%20PR.%20Could%20you%20please%20have%20a%20look%20and%20see%20if%20it%20make%20sense%3F%20I%27m%20still%20not%20completely%20happy%20with%20it%2C%20so%20feel%20free%20to%20improve.%22%2C%20%22created_at%22%3A%20%222015-11-19T10%3A59%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40gehel%20I%27ll%20do%20another%20commit%20with%20some%20of%20your%20ideas%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-11-19T12%3A41%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1846171%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gmcoringa%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b18ea1e36571867f69c559151235ea72b979d3c0%20jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java%2071%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/363%23discussion_r45325132%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20use%20placeholders%20and%20remove%20the%20%60if%20%28isDebugEnabled%28%29%29%60.%20See%20%5Bslf4j%20manual%5D%28http%3A//www.slf4j.org/manual.html%23typical_usage%29%20for%20details.%22%2C%20%22created_at%22%3A%20%222015-11-19T10%3A43%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java%3AL195-205%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull b18ea1e36571867f69c559151235ea72b979d3c0 jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java 71'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/363#discussion_r45325132'>File: jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java:L195-205</a></b>
- <a href='https://github.com/gehel'></a> Please use placeholders and remove the `if (isDebugEnabled())`. See [slf4j manual](http://www.slf4j.org/manual.html#typical_usage) for details.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/363#issuecomment-158023094'>General Comment</a></b>
- <a href='https://github.com/gehel'></a> I sent you some ideas on what I think is cleaner code via another PR. Could you please have a look and see if it make sense? I'm still not completely happy with it, so feel free to improve.
- <a href='https://github.com/gmcoringa'></a> @gehel I'll do another commit with some of your ideas :)

<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/363?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/363?approve=1'></a>&nbsp;<a href='https://github.com/jmxtrans/jmxtrans/pull/363'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by gehel** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/jmxtrans/jmxtrans/pull/363'>Refresh</a>.<a href='#crh-update-none'></a>

File: jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java [hunk 1]
```diff
@@ -180,6 +195,10 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 
 	private synchronized boolean doSend(String stat) {
 		try {
+			if (isDebugEnabled()) {
```
Comment (gehel) [inline jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java]: Please use placeholders and remove the `if (isDebugEnabled())`. See [slf4j manual](http://www.slf4j.org/manual.html#typical_usage) for details.

**Comment by gehel** [comment]
I sent you some ideas on what I think is cleaner code via another PR. Could you please have a look and see if it make sense? I'm still not completely happy with it, so feel free to improve.

**Comment by gmcoringa** [comment]
@gehel I'll do another commit with some of your ideas :)

**Comment by gehel** [comment]
Looks good to me. I'll do some more invasive refactoring and we should be able to add unit tests for this new functionality.

**Comment by gehel** [comment]
If you are interested, I did some more refactoring and added some tests in #364. I actually created another StatsDWriter as I do not want to force that change on everyone just yet.

**Pull request merged**